go/adbc/driver/snowflake/connection.go

```diff
@@ -85,82 +127,165 @@ func (c *connectionImpl) GetObjects(ctx context.Context, depth adbc.ObjectDepth,
 	defer conn.Close()
 
 	gQueryIDs, gQueryIDsCtx := errgroup.WithContext(ctx)
+
 	queryFile := queryTemplateGetObjectsAll
 	switch depth {
 	case adbc.ObjectDepthCatalogs:
-		if catalog == nil {
-			queryFile = queryTemplateGetObjectsTerseCatalogs
-			// if the catalog is null, show the terse databases
-			// which doesn't require a database context
-			gQueryIDs.Go(func() error {
-				return conn.Raw(func(driverConn any) error {
-					rows, err := driverConn.(driver.QueryerContext).QueryContext(gQueryIDsCtx, "SHOW TERSE DATABASES", nil)
-					if err != nil {
-						return err
-					}
+		queryFile = queryTemplateGetObjectsTerseCatalogs
+		gQueryIDs.Go(func() error {
+			return conn.Raw(func(driverConn any) (err error) {
+				query := "SHOW TERSE /* ADBC:getObjectsCatalogs */ DATABASES"
+				if catalog != nil && len(*catalog) > 0 && *catalog != "%" && *catalog != ".*" {
+					query += " LIKE '" + escapeSingleQuoteForLike(*catalog) + "'"
```

Comment:
ah, does Snowflake not allow bind parameters here?

Reply:
in my testing, that was the case. My original attempt was to use bind parameters but it didn't pick up the values (it didn't error, just was a blank string)